src/Nethermind/Nethermind.Core/Crypto/Keccak.cs

```diff
@@ -101,30 +111,30 @@ public static ValueKeccak Compute(ReadOnlySpan<byte> input)
                 return OfAnEmptyString;
             }
 
-            ValueKeccak result = default;
-            KeccakHash.ComputeHashBytesToSpan(input, result.BytesAsSpan);
-            return result;
+            Span<byte> span = stackalloc byte[MemorySize];
+            KeccakHash.ComputeHashBytesToSpan(input, span);
```

Comment:
Would it make any difference to create a `ValueKeccak` first, expose a `Span<byte>` (not read only), and use that as destination instead?

Reply:
Tried that; it got very inconstant in actually updating the `ValueKeccak`. I assume since its both a `readonly` struct and a 32byte register type the Jit makes assumptions and applies optimizations which means it doesn't always write back

Reply:
Noted, can you add a comment on this?

Reply:
🤔 I was thinking of a different place where modified after construction.

Let me take a look at this one again

Reply:
Looks good!